src/Shared/Buffers.MemoryPool/DefaultMemoryPoolFactory.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.AspNetCore.Connections;
+
+namespace Microsoft.AspNetCore;
+
+#nullable enable
+
+internal sealed class DefaultMemoryPoolFactory : IMemoryPoolFactory<byte>, IAsyncDisposable
+{
+    private readonly IMeterFactory _meterFactory;
+    private readonly ConcurrentDictionary<PinnedBlockMemoryPool, PinnedBlockMemoryPool> _pools = new();
+    private readonly PeriodicTimer _timer;
+    private readonly Task _timerTask;
+
+    public DefaultMemoryPoolFactory(IMeterFactory? meterFactory = null)
+    {
+        _meterFactory = meterFactory ?? NoopMeterFactory.Instance;
+        _timer = new PeriodicTimer(TimeSpan.FromSeconds(10));
+        _timerTask = Task.Run(async () =>
+        {
+            try
+            {
+                while (await _timer.WaitForNextTickAsync())
+                {
+                    foreach (var pool in _pools.Keys)
+                    {
+                        pool.PerformEviction();
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine(ex);
+            }
+        });
+    }
+
+    public MemoryPool<byte> Create()
+    {
+        var pool = new PinnedBlockMemoryPool(_meterFactory);
+
+        _pools.TryAdd(pool, pool);
+
+        pool.OnPoolDisposed(static (state, self) =>
+        {
+            ((ConcurrentDictionary<PinnedBlockMemoryPool, PinnedBlockMemoryPool>)state!).TryRemove(self, out _);
+        }, _pools);
+
+        return pool;
+    }
+
+    public async ValueTask DisposeAsync()
+    {
+        _timer.Dispose();
+        await _timerTask;
+    }
+
+    private sealed class NoopMeterFactory : IMeterFactory
```

Comment:
This name isn't accurate because the factory will still create meters and people can still listen to those meters. The only difference is the factory isn't set as the owner of the meters. In other words, they're harder to test.

A better simulation of "no op" would be to not create `PinnedBlockMemoryPoolMetrics` if there's no meter factory and any calls to it would be `_metrics?.UpdateCurrentMemory(-block.Memory.Length);`

Reply:
I went with allowing a null `IMeterFactory`. Most tests aren't doing anything with meters, so it seems like a waste to require them to pass in a meter factory.